File: x/wasmx/keeper/testdata/python/call.py
import json
from wasmx import call, call_static

def instantiate():
    pass

def main(input):
    if "store" in input:
        return wrapStore(*input["store"])
    if "load" in input:
        return wrapLoad(*input["load"])
    raise ValueError('Invalid function')

def wrapStore(address, value):
    calldata = json.dumps({"store":[value]})
    res = call(1000000, address, 0, calldata.encode())
    response = json.loads(res.decode())
    return response["data"]

def wrapLoad(address):
    calldata = json.dumps({"load":[]})
    res = call_static(1000000, address, calldata.encode())
    response = json.loads(res)
    data = response["data"]
    return bytes(data) + b'23'

File: x/wasmx/keeper/testdata/js/call.js
import * as wasmx from 'wasmx';

export function instantiate() {}

export function main(dataObj) {
    if (dataObj.store) {
        return wrapStore(...dataObj.store);
    } else if (dataObj.load) {
        return wrapLoad(...dataObj.load);
    }
    throw new Error("invalid function");
}

function wrapStore(address, value) {
    let calldata = JSON.stringify({"store":[value]})
    return wasmx.call(1000000, address, new ArrayBuffer(32), stringToArrayBuffer(calldata))
}

function wrapLoad(address) {
    let calldata = JSON.stringify({"load":[]})
    let res = wasmx.callStatic(1000000, address, stringToArrayBuffer(calldata))
    let response = JSON.parse(arrayBufferToString(res))
    let data = new Uint8Array(Object.values(response.data));
    return data.buffer;
}

function stringToArrayBuffer(inputString) {
    const bytes = new Uint8Array(inputString.length);
    for (let i = 0; i < inputString.length; i++) {
        bytes[i] = inputString.charCodeAt(i) & 0xFF;
    }
    return bytes.buffer;
}

function arrayBufferToString(arrayBuffer) {
    const bytes = new Uint8Array(arrayBuffer);
    let result = "";
    for (let i = 0; i < bytes.length; i++) {
        result += String.fromCharCode(bytes[i]);
    }
    return result;
}
